src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -5322,124 +5258,94 @@ def _set_log_analytics_workspace_extension(cmd, resource_group_name, vm, vm_name
-                                            rotation_to_latest_key_version_enabled=enable_auto_key_rotation)
-
-    if federated_client_id is not None:
-        disk_encryption_set.federated_client_id = federated_client_id
-
-    return sdk_no_wait(no_wait, client.begin_create_or_update, resource_group_name, disk_encryption_set_name,
-                       disk_encryption_set)
+def assign_disk_encryption_set_identity(cmd, resource_group_name, disk_encryption_set_name,
+                                        mi_system_assigned=None, mi_user_assigned=None):
+    from .aaz.latest.disk_encryption_set import Update as _Update, Show as _Show
 
+    args = {
+        "disk_encryption_set_name": disk_encryption_set_name,
+        "resource_group": resource_group_name
+    }
+    identity = _Show(cli_ctx=cmd.cli_ctx)(command_args=args).get('identity', {})
 
-def update_disk_encryption_set(cmd, instance, client, resource_group_name, key_url=None, source_vault=None,
-                               enable_auto_key_rotation=None, federated_client_id=None):
-    from azure.mgmt.core.tools import resource_id, is_valid_resource_id
-    from azure.cli.core.commands.client_factory import get_subscription_id
-    if key_url:
-        instance.active_key.key_url = key_url
+    existing_system_identity = False
+    existing_user_identities = set()
+    if identity:
+        existing_system_identity = identity['type'] in ['SystemAssigned', 'SystemAssigned, UserAssigned']
+        existing_user_identities = {x.lower() for x in list(identity.get('userAssignedIdentities', {}).keys())}
 
-    if source_vault:
-        if not is_valid_resource_id(source_vault):
-            source_vault = resource_id(subscription=get_subscription_id(cmd.cli_ctx),
-                                       resource_group=resource_group_name,
-                                       namespace='Microsoft.KeyVault', type='vaults', name=source_vault)
-        instance.active_key.source_vault = {'id': source_vault}
+    add_system_assigned = mi_system_assigned
+    add_user_assigned = {x.lower() for x in (mi_user_assigned or [])}
 
-    if enable_auto_key_rotation is not None:
-        instance.rotation_to_latest_key_version_enabled = enable_auto_key_rotation
+    updated_system_assigned = existing_system_identity or add_system_assigned
+    updated_user_assigned = list(existing_user_identities.union(add_user_assigned))
 
-    if federated_client_id is not None:
-        instance.federated_client_id = federated_client_id
+    identity['type'] = 'SystemAssigned'
+    if updated_user_assigned:
+        identity['type'] = 'SystemAssigned, UserAssigned' if updated_system_assigned else 'UserAssigned'
+        identity['userAssignedIdentities'] = dict.fromkeys(updated_user_assigned, {})
 
-    return instance
+    args['identity'] = identity
+    result = LongRunningOperation(cmd.cli_ctx)(_Update(cli_ctx=cmd.cli_ctx)(command_args=args))
```

Comment:
Since the `args` is generated from the output of the show command and it's camelcase. I'm not sure if it works.